wcfsetup/install/files/acp/install_com.woltlab.wcf_step2.php

```
@@ -24,22 +25,38 @@
 ]);
 
 // install default user ranks
+$sql = "INSERT INTO wcf1_user_rank_content
+                    (rankID, languageID, title)
+        VALUES      (?, ?, ?)";
+$statement = WCF::getDB()->prepare($sql);
+
 foreach ([
-    [4, 0, 'wcf.user.rank.administrator', 'blue'],
-    [5, 0, 'wcf.user.rank.moderator', 'blue'],
-    [3, 0, 'wcf.user.rank.user0', ''],
-    [3, 300, 'wcf.user.rank.user1', ''],
-    [3, 900, 'wcf.user.rank.user2', ''],
-    [3, 3000, 'wcf.user.rank.user3', ''],
-    [3, 9000, 'wcf.user.rank.user4', ''],
-    [3, 15000, 'wcf.user.rank.user5', ''],
-] as [$groupID, $requiredPoints, $rankTitle, $cssClassName]) {
-    UserRankEditor::create([
+    [4, 0, ['de' => 'Administrator', 'en' => 'Administrator'], 'blue'],
+    [5, 0, ['de' => 'Moderator', 'en' => 'Moderator'], 'blue'],
+    [3, 0, ['de' => 'Anfänger', 'en' => 'Beginner'], ''],
+    [3, 300, ['de' => 'Schüler', 'en' => 'Student'], ''],
+    [3, 900, ['de' => 'Fortgeschrittener', 'en' => 'Intermediate'], ''],
+    [3, 3000, ['de' => 'Profi', 'en' => 'Professional'], ''],
+    [3, 9000, ['de' => 'Meister', 'en' => 'Master'], ''],
+    [3, 15000, ['de' => 'Erleuchteter', 'en' => 'Enlightened'], ''],
+] as [$groupID, $requiredPoints, $rankTitles, $cssClassName]) {
+    $userRank = UserRankEditor::create([
         'groupID' => $groupID,
         'requiredPoints' => $requiredPoints,
-        'rankTitle' => $rankTitle,
         'cssClassName' => $cssClassName,
     ]);
+
+    foreach (LanguageFactory::getInstance()->getLanguages() as $language) {
+        if (!isset($rankTitles[$language->languageCode])) {
+            continue;
+        }
+
```

```suggestion
```

The languages are fixed and known at this time, I rather have the code fail hard than to make incorrect assumptions.